enable/abstract_window.py

```diff
@@ -218,25 +218,26 @@ def _component_changed(self, old, new):
         # If possible, size the new component according to the size of the
         # toolkit control
         size = self._get_control_size()
-        if (size is not None) and hasattr(self.component, "bounds"):
-            new.on_trait_change(self.component_bounds_changed, "bounds")
-            pix_scale = self.base_pixel_scale
-            if getattr(self.component, "fit_window", False):
-                self.component.outer_position = [0, 0]
-                self.component.outer_bounds = [
-                    size[0] / pix_scale, size[1] / pix_scale
-                ]
-            elif hasattr(self.component, "resizable"):
-                if "h" in self.component.resizable:
-                    self.component.outer_x = 0
-                    self.component.outer_width = size[0] / pix_scale
-                if "v" in self.component.resizable:
-                    self.component.outer_y = 0
-                    self.component.outer_height = size[1] / pix_scale
+        if hasattr(self.component, "bounds"):
```

Comment:
I believe this was previously a bug, as there is a case in which size is None when a component is `new`, but later the method is called with that same component as `old`.  We were trying then to remove a listener that had never gotten hooked up.  `observe` complained about this and the test suite failed, with `on_trait_change` it flew under the radar

Reply:
I just added a unit test which exercises this part of the code in #635. Can you add a test which would break if this change weren't here?

Reply:
I've added a test now.  It previously failed, and is now passing (so in that sense it is good). The test itself feels a bit ugly to me though in general :/ I'm very open to suggested modifications